src/signaling/common.ts

```diff
@@ -191,7 +189,45 @@ export abstract class Signaling implements saltyrtc.Signaling {
     }
 
     /**
-     * Return connection path for websocket.
+     * Close the WebSocket connection.
+     *
+     * @param code The close code.
+     * @param reason The close reason.
+     * @param unbind Whether to unbind all events. This will move the Signaling
+     *   instance into the `closed` state.
+     */
+    private closeWebsocket(code?: number, reason?: string, unbind = false): void {
+        if (this.ws !== null) {
+            // Drop internal close codes
+            // see: https://github.com/saltyrtc/saltyrtc-meta/issues/110
+            if (code === undefined || code <= 3000) {
+                code = CloseCode.ClosingNormal;
+            }
+
+            // Disconnect
+            console.debug(this.logTag, `Disconnecting WebSocket, close code: ${code}`);
+            this.ws.close(code, reason);
+
+            // Unbind events?
+            if (unbind) {
+                this.ws.removeEventListener('open', this.onOpen);
+                this.ws.removeEventListener('error', this.onError);
+                this.ws.removeEventListener('close', this.onClose);
+                this.ws.removeEventListener('message', this.onMessage);
+            }
+
+            // Forget instance
+            this.ws = null;
+
+            // Move into closed state (if necessary)
+            if (unbind) {
+                this.setState('closed');
+            }
```

Comment:
Why is this only being done when `unbind` is set?

Reply:
Ah, because this is done by default in the `onClose` implementation.

It's important that this is documented in the docstring of the public method.

Reply:
Which one is missing documentation? I'm a bit lost. :)